shumai/tensor/register_gradients.ts

```diff
@@ -5,11 +5,19 @@ const sm = { ...base, ...ops }
 
 export interface Grad {
   idx: number
-  in: Tensor[]
+  in: any[]
   grad_in: Tensor
   out: Tensor
 }
 
+function getBackwardGradient(grad: Grad): Tensor {
```

Comment:
nit: is this going to be used outside `concatenate`'s gradient? may be worth inlining?